### ISSUE TITLE
Following the example results in "No possible anchors found"

### ISSUE THREAD

**Issue by k3rn3l3rr0r** [dual-wield-ray/PyDrumScore#60]
Hi and thanks for creating this!

After installing through pip in fresh venv the `pydrumscore basic_beat` command crashes because there is no .git folder present.
I see the TODO in get_project_root.py ... :)  "# TODO(ekon): add warning about relying on ".git"" so this is known but it is currently not mentioned in the docs.

Anyway, to get pydrumscore command to work I had to change the config_root in export.py by hand to point to the right directory (used absolute path just to try it out). Because even if I created the user config.ini in the root directory the command kept crashing on `assert default_configur.has_option(section, option)` as the default_configur was not loaded and set.

**Comment by dual-wield-ray** [comment]
Looking at this now, so in order to save time/mistakes I use a small library called "[from-root](https://pypi.org/project/from-root/)" to get root dirs. In the get_project_root.py it searches first for path that contains the string "site-packages", ie. the folder where pip installs your packages by default. Then if it doesn't find this, it assumes you are on a git clone, so it moves upwards until it finds a .git folder. And if it doesn't find _that_, it throws the error you got.

On a fresh venv, I have a path like this: `.../PyDrumScore-User/.venv/Lib/site-packages/pydrumscore`. If I comment the section that searches for "site-packages" I get the same error. Is it possible that your packages are installed in a directory with a different name?

**Comment by k3rn3l3rr0r** [comment]
:+1: 
My workflow here was like this - I just opened my PyCharm and created new project "drumScoring" and let PyCharm create new empty venv for me. Then I created requirements.txt with only PyDrumScore in it and installed dependencies when prompted about the changes.

The resulting installation path is this:
`.../drumScoring/venv/lib/python3.10/site-packages/pydrumscore`

I've now tried the same process but with the project (and folder) named pydrumscore, just to be sure, and the result is the same. As in failing on _get_config_option because no config file could be found and loaded.

**Comment by dual-wield-ray** [comment]
Ok I have a repro, looks like the cause is the way PyCharm handles installs. It seems that it ignores the package version in the requirement.txt (from-root==1.0.2) and instead gets the latest (1.2.0 atm). That version doesn't have the "site-packages" section I was talking about earlier (not sure why when it worked fine). I use VSCode and it fetches me the correct version. So maybe there's something wrong in the package metadata that PyCharm needs to work, in the meantime force-installing the "from-root" version to 1.0.2 seems like a cleaner workaround then doing local code changes.

**Comment by k3rn3l3rr0r** [comment]
Yes, it is true - looking into the venv libs, the installed from_root version is indeed 1.2.0 even if your requirements.txt are specifying 1.0.2 ... That's, not nice.

**Comment by dual-wield-ray** [comment]
So there's a bit of shenanigans regarding packaging setups for python, basically there's the one you put in requirements.txt and there's the ones you put in the setup.py/pyproject.toml. People sometimes copy-paste in both places but that's a bad practice. Long explanation is here: https://caremad.io/posts/2013/07/setup-vs-requirement/
TLDR, I think we just need to pin the deps in the setup.py instead of the requirements.txt, which is ok to do because PyDrumScore is only a library and not a full application. Which will then write it in the METADATA file in the package, and I suspect that file is what PyCharm pulls dependencies from.

**Comment by k3rn3l3rr0r** [comment]
Yes, that sounds logical. Thanks for the explanation.

**Comment by dual-wield-ray** [comment]
Fix in 0.0.6

(Also asked the dev about the removal of the logic [here].(https://github.com/EduardKononov/from-root/issues/1))
